Hyper/Meshes/Chunk.cs

```diff
@@ -0,0 +1,133 @@
+﻿using Hyper.MarchingCubes;
+using NLog;
+using OpenTK.Graphics.OpenGL4;
+using OpenTK.Mathematics;
+
+namespace Hyper.Meshes
+{
+    internal class Chunk : Mesh
+    {
+        public const int Size = 32;
+
+        public new Vector3i Position;
+
+        public Vector3i Middle => Position + Vector3i.One * Size / 2;
+
+        private float[,,] _voxels;
+
+        private static readonly Logger _logger = LogManager.GetCurrentClassLogger();
+
+        public Chunk(float[] vertices, Vector3i position, float[,,] voxels) : base(vertices, position)
+        {
+            _voxels = voxels;
+            Position = position;
+            base.Position = position; // TODO ugly as hell
```

Comment:
this line does not do anything since the base constructor already sets this parameter.
Agree that it's ugly. Maybe something like this? 
        public new Vector3i Position { get => new Vector3i((int)base.Position.X, (int)base.Position.Y, (int)base.Position.Z); set => base.Position = value;}

Reply:
Just removing this line is also ok since i dont think we will ever change the position

Reply:
I didn't notice it was being set in the base class constructor